### PR TITLE
Check isSemanticdbLocal and assign LOCAL flag accordingly

### PR DESCRIPTION
**Pull request by andyscott**
Scala Spree hacking session with @olafurpg! This should address #1486.

### CODE REVIEW

File: tests/jvm/src/test/scala/scala/meta/tests/semanticdb/TargetedSuite.scala [hunk 1]
```diff
@@ -4,6 +4,7 @@ package semanticdb
 import org.langmeta.internal.semanticdb._
 import scala.meta._
 import scala.meta.internal.semanticdb.scalac._
+import scala.meta.internal.semanticdb3.SymbolInformation
```
Comment (olafurpg) [inline tests/jvm/src/test/scala/scala/meta/tests/semanticdb/TargetedSuite.scala]: For consistency with the rest of the codebase
```scala
 import scala.meta.internal.semanticdb3.SymbolInformation.{Kind => k}
```

File: tests/jvm/src/test/scala/scala/meta/tests/semanticdb/TargetedSuite.scala [hunk 2]
```diff
@@ -1315,4 +1316,20 @@ class TargetedSuite extends DatabaseSuite(SemanticdbMode.Slim) {
...
+      |  for {
+      |    i <- List(1, 2)
+      |    <<j>> <- List(3, 4)
+      |  } yield j
+      |}
+      """.stripMargin, { (db, j) =>
+      val denot = db.symbols.find(_.symbol == j).get.denotation
+      val kind = denot.skind
+
+      assert(kind == k.LOCAL)
```
Comment (gabro) [inline tests/jvm/src/test/scala/scala/meta/tests/semanticdb/TargetedSuite.scala]: For consistency with https://github.com/scalameta/scalameta/pull/1501#discussion_r181561316, maybe make this `k.isLocal`?

Comment (olafurpg) [reply tests/jvm/src/test/scala/scala/meta/tests/semanticdb/TargetedSuite.scala]: That's a good point, I agree. Gonna merge nevertheless so @andyscott can get a t-shirt 😂

Comment (gabro) [reply tests/jvm/src/test/scala/scala/meta/tests/semanticdb/TargetedSuite.scala]: I won't stand between a man and his t-shirt 😂

Comment (andyscott) [reply tests/jvm/src/test/scala/scala/meta/tests/semanticdb/TargetedSuite.scala]: 👕👍

**Pull request merged**